pkg/cmd/sync/tasks.go

```diff
@@ -5,35 +5,99 @@ import (
 	"fmt"
 
 	msg "github.com/aziontech/azion-cli/messages/sync"
-	api "github.com/aziontech/azion-cli/pkg/api/edge_applications"
+	edgeApp "github.com/aziontech/azion-cli/pkg/api/edge_applications"
+	"github.com/aziontech/azion-cli/pkg/api/origin"
 	"github.com/aziontech/azion-cli/pkg/cmdutil"
 	"github.com/aziontech/azion-cli/pkg/contracts"
 	"github.com/aziontech/azion-cli/pkg/logger"
 	"go.uber.org/zap"
 )
 
 var (
-	C    context.Context
-	Opts *contracts.ListOptions
+	opts *contracts.ListOptions
+	ctx  context.Context = context.Background()
 )
 
 func (synch *SyncCmd) SyncResources(f *cmdutil.Factory, info contracts.SyncOpts) error {
-	C = context.Background()
-	Opts = &contracts.ListOptions{
+	opts = &contracts.ListOptions{
 		PageSize: 1000,
 		Page:     1,
 	}
-	err := synch.syncRules(info, f)
+
+	var err error
+	err = synch.syncRules(info, f)
+	if err != nil {
+		return fmt.Errorf(msg.ERRORSYNC, err.Error())
+	}
+
+	err = synch.syncCache(info, f)
 	if err != nil {
 		return fmt.Errorf(msg.ERRORSYNC, err.Error())
 	}
+
+	err = synch.syncOrigin(info, f)
+	if err != nil {
+		return fmt.Errorf(msg.ERRORSYNC, err.Error())
+	}
+
 	return nil
 }
 
-func (synch *SyncCmd) syncRules(info contracts.SyncOpts, f *cmdutil.Factory) error {
+func (synch *SyncCmd) syncOrigin(info contracts.SyncOpts, f *cmdutil.Factory) error {
+	client := origin.NewClient(f.HttpClient, f.Config.GetString("api_url"), f.Config.GetString("token"))
+	resp, err := client.ListOrigins(ctx, opts, info.Conf.Application.ID)
+	if err != nil {
+		return err
+	}
+	for _, origin := range resp.Results {
+		fmt.Println(origin)
+		if r := info.OriginIds[origin.Name]; r.OriginId > 0 || origin.Name == "Default Rule" {
+			continue
+		}
+		newOrigin := contracts.AzionJsonDataOrigin{
+			OriginId:  origin.GetOriginId(),
+			OriginKey: origin.GetOriginKey(),
+			Name:      origin.GetName(),
+		}
+		info.Conf.Origin = append(info.Conf.Origin, newOrigin)
+		err := synch.WriteAzionJsonContent(info.Conf, ProjectConf)
+		if err != nil {
+			logger.Debug("Error while writing azion.json file", zap.Error(err))
+			return err
+		}
+		logger.FInfo(synch.Io.Out, fmt.Sprintf(msg.SYNCMESSAGEORIGIN, origin.Name))
+	}
+	return nil
+}
+
+func (synch *SyncCmd) syncCache(info contracts.SyncOpts, f *cmdutil.Factory) error {
+	client := edgeApp.NewClient(f.HttpClient, f.Config.GetString("api_url"), f.Config.GetString("token"))
+	resp, err := client.ListCacheEdgeApp(context.Background(), info.Conf.Application.ID)
+	if err != nil {
+		return err
+	}
+	for _, cache := range resp {
+		if r := info.CacheIds[cache.Name]; r.Id > 0 || cache.Name == "Default Rule" {
```

Comment:
do we need to check for a cache named "Default Rule"?

Reply:
Kkkk Copy 👍🏼

Reply:
thought so 😆